src/dmd/e2ir.d

```diff
@@ -268,6 +268,52 @@ private elem *callfunc(const ref Loc loc,
                  */
                 ea.Ety = TYllong;
             }
+
+            /* Do integral promotions. This is not necessary per the C ABI, but
+             * some code from the C world seems to rely on it.
+             */
+            if (op == NotIntrinsic && tyintegral(ea.Ety) && arg.type.size(arg.loc) < 4)
+            {
+                if (ea.Eoper == OPconst)
+                {
+                    ea.EV.Vullong = el_tolong(ea);
+                    ea.Ety = TYint;
+                }
+                else
+                {
+                    OPER opc;
+                    switch (tybasic(ea.Ety))
+                    {
+                        case TYbool:
+                        case TYchar:
+                        case TYuchar:
+                        case TYchar8:
+                            opc = OPu8_16;
```

Comment:
Shouldn't this be OPu8_16 followed by OPu16_32?

Reply:
No, because OPu8_16 will convert to 32 in 32/64 bit code generation.